specs/core/0_beacon-chain.md

```diff
@@ -1596,41 +1611,35 @@ _Note_: If there are skipped slots between a block and its parent block, run the
 
 ### Per-slot processing
 
-Below are the processing steps that happen at every slot.
-
-#### Slot
+Below are the processing steps that happen at every `slot > GENESIS_SLOT`.
 
+* Set `state.latest_state_roots[state.slot % SLOTS_PER_BATCHING] = hash(state)`.
+* Set `state.latest_block_roots[state.slot % SLOTS_PER_BATCHING] = get_block_root(state, state.slot)`.
+* If `state.slot % SLOTS_PER_BATCHING == 0` append `merkle_root(state.latest_block_roots + state.latest_state_roots)` to `state.historical_batchings`.
```

Comment:
Should we use `hash_tree_root` on the concatenated list instead?

Reply:
The reason I went with `merkle_root` is that if we consider the concatenation as a `List` then we need to include the list size in the `hash_tree_root`. Here we have a fixed size data structure (let's call that an "array") so including the size is a nuisance. It's the same problem we have with shard data.

This is arguably a bug with `hash_tree_root`. We already support byte arrays (fixed size) I'll file an issue to generalise friendly Merkleisation to arbitrary arrays. 👍

Reply:
I'd say `merkle_root` is fine here.